server/src/main/java/org/cloudfoundry/identity/uaa/authentication/manager/PasswordGrantAuthenticationManager.java

```diff
@@ -132,9 +134,10 @@ Authentication oidcPasswordGrant(Authentication authentication, OIDCIdentityProv
         if (clientId == null) {
             throw new ProviderConfigurationException("External OpenID Connect provider configuration is missing relyingPartyId.");
         }
-        if (clientSecret == null && config.getJwtClientAuthentication() == null) {
-            throw new ProviderConfigurationException("External OpenID Connect provider configuration is missing relyingPartySecret or jwtclientAuthentication.");
+        if (clientSecret == null && config.getJwtClientAuthentication() == null && config.getAuthMethod() == null) {
```

Comment:
Is this not a breaking change?

Reply:
no because we have as default config.authMethod = null, means, if you do not provide any explicit method, and you dont have a secret or key configured, then if fails.

However, if you setup config.authMethod = none, then you should have the option to omit a secret (client authentication for the resource owner flow / password grant), because we have examples of OIDC providers, that allow this, e.g. https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth-ropc (Entra / Azure) ... see client_secret / client_assertion section. 
Btw. Okta and also SAP OIDC IDP and others provide this, therefore we should allow it.